src/main/java/net/io/kino/service/impl/MovieServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package net.io.kino.service.impl;
+
+import net.io.kino.model.Movie;
+import net.io.kino.repository.MovieRepository;
+import net.io.kino.service.MovieService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class MovieServiceImpl implements MovieService {
+
+    @Autowired
+    MovieRepository movies;
+
+    @Override
+    public Movie createMovie(Movie movie) {
+        List<Movie> existingMovies = movies.findMovieByTitle(movie.getTitle());
+        if (existingMovies.size() > 0)
+        {
+            throw new IllegalArgumentException("This movie already exists in repository.");
+        }
+        return movies.save(movie);
+    }
+
+    @Override
+    public Movie getMovieById(long id){
+        return (Movie) movies.findMovieById(id);
```

Comment:
Kara chłosty za castowanie `List<Movie>` na Movie

Reply:
Tutaj też można użyć findById żeby zwrócić jeden film, bo id jest unikalne to zawsze zwróci jeden element